pkg/initializer/initializer.go

```diff
@@ -71,20 +71,30 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 		if err != nil {
 			logger.Fatalf("failed to create clientset, %v", err)
 		}
+
 		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
-		isScaleup, err := m.IsClusterScaledUp(ctx, clientSet)
-		if err != nil {
-			logger.Errorf("scale-up not detected: %v", err)
-		} else if isScaleup {
-			logger.Info("Etcd cluster scale-up is detected")
-			// Add a learner(non-voting member) to a etcd cluster with retry
-			// If backup-restore is unable to add a learner in a cluster
-			// restart the `initialization` by exiting the backup-restore.
-			if err := member.AddLearnerWithRetry(ctx, m, addLearnerAttempts, e.Config.RestoreOptions.Config.DataDir); err != nil {
-				logger.Fatalf("unable to add a learner in a cluster: %v", err)
+
+		// check heartbeat of etcd member
+		if memberHearbeatPresent, err = m.WasMemberInCluster(ctx, clientSet); memberHearbeatPresent {
+			logger.Info("member found to be already a part of the cluster")
+			logger.Info("skipping the scale-up check")
```

Comment:
The error is not handled here. You can just `log` the error and go to `IsClusterScaledUp` check.

Reply:
If `memberHearbeatPresent` is `false` or `err!=nil` , in both cases we move to `else{ ... }` part.
ok, If you want then I can just log the error. Hope that's fine.